app/DTO/UserWithCapabilitiesDTO.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\DTO;
+
+use App\Models\User as UserModel;
+
+/**
+ * Data Transfer Object (DTO) to transmit the capabilities of a User.
+ *
+ * This is used in the Admin part to manage users.
+ *
+ * If we return directly the User (e.g. for sharing), we are given only the following attributes:
+ * - id
+ * - username
+ * - email
+ * - has_token
+ *
+ * This is because we do not send more than the front-end needs to know.
```

Comment:
```suggestion
 * Data Transfer Object (DTO) to transmit the capabilities of a user.
 *
 * This DTO is used by the admin part to manage users.
 *
 * If we directly return the user (e.g. for sharing), we only transmit the following attributes:
 * - id
 * - username
 * - email
 * - has_token
 *
 * This is because we do not send more than the front-end needs to know.
```

Although I better understand the reason for this DTO now, I wonder if a) it _should_ be this way and b) why it is this way. I checked with the current code. `UserController::list()` currently returns `User::query()->where('id', '>', 0)->get()`. Hence, I don't (yet) understand the statement that a directly returned user object only transmits a subset of the attributes. Is this another change of this PR? Maybe I need to read on and I will find the answer below.